lib/helper/Testcafe.js

```diff
@@ -0,0 +1,661 @@
+// @ts-nocheck
+const fs = require('fs');
+const assert = require('assert');
+const path = require('path');
+const requireg = require('requireg');
+const createTestCafe = require('testcafe');
+const { Selector, ClientFunction } = require('testcafe');
+
+const testControllerHolder = require('./testcafe/testControllerHolder');
+const { mapError, createTestFile } = require('./testcafe/testcafe-utils');
+
+const stringIncludes = require('../assert/include').includes;
+const { urlEquals } = require('../assert/equal');
+const {
+  xpathLocator,
+} = require('../utils');
+const Locator = require('../locator');
+const Helper = require('../helper');
+
+/**
+ * Client Functions
+ */
+const getPageUrl = t => ClientFunction(() => document.location.href).with({ boundTestRun: t });
+
+/**
+ * Uses Testcafe to run tests.
+ */
+class Testcafe extends Helper {
+  constructor(config) {
+    super(config);
+
+    this.iteration = 1;
+    this.testcafe = undefined; // testcafe instance
+    this.t = undefined; // testcafe test controller
+    this.dummyTestcafeFile; // generated testcafe test file
+
+    this.context = undefined; // TODO Not sure if this applies to testcafe
+
+
+    this.options = Object.assign({
+      url: 'http://localhost',
+      show: false,
+      browser: 'chrome',
+      restart: true, // TODO Test if restart false works
+      manualStart: false,
+      keepBrowserState: false,
+      waitForTimeout: 10000,
+      fullPageScreenshots: false,
+      disableScreenshots: false,
+      windowSize: undefined,
+    }, config);
+  }
+
+  // TOOD Do a requirements check
+  static _checkRequirements() {
+    try {
+      requireg('testcafe');
+    } catch (e) {
+      return ['testcafe@^1.1.0'];
+    }
+  }
+
+  async _startBrowser() {
+    this.dummyTestcafeFile = createTestFile(); // create a dummy test file to get hold of the test controller
```

Comment:
Can we have it without extra files? By just calling TestCafe API ?

Reply:
I dont see how. They don't have a real API. The only thing you can do is run testcafe test files. Therefore this trick. Got it from here https://github.com/rquellh/testcafe-cucumber